code/src/components/App.jsx

```diff
@@ -562,20 +564,24 @@ export default class App extends React.Component {
 
   onLayerIdChange = (index, oldId, newId) => {
     const opts = {};
+    const changedStyle = { ...this.state.mapStyle };
+
     let selectableLayerIndex = 0;
-    const changedStyle = {...this.state.mapStyle}
+
     changedStyle.layers.forEach(layer => {
-      if (isLayerSelectable(layer)) {
-        if (selectableLayerIndex == index) {
-          if (this.state.selectedLayerOriginalId == layer.id) {
-            opts.newSelectedLayerId = newId
-          }
-          layer.id = newId
+      if (!isLayerSelectable(layer)) return;
+
+      if (selectableLayerIndex === index) {
+        if (this.state.selectedLayerOriginalId === layer.id) {
+          opts.newSelectedLayerId = newId;
         }
-        ++selectableLayerIndex;
```

Comment:
What is this for? Ok to remove it?

Reply:
I just refactored the code. You can see that I have the same code at L581.